pyOCD/target/target_tz10xx.py

```diff
@@ -0,0 +1,352 @@
+"""
+TOSHIBA TZ10xx serias
+"""
+from .coresight_target import CoreSightTarget
+from .memory_map import (FlashRegion, RamRegion, MemoryMap)
+from time import sleep
+
+class TZ10xx(CoreSightTarget):
+
+    has_fpu = True
+    memoryMap = MemoryMap(
+        FlashRegion(    start=0,           length=0x00100000,   blocksize=0x100, isBootMemory=True),   #On package NOR Flash
+        RamRegion(      start=0x10000000,  length=0x00040000),                                          #Code region
+        RamRegion(      start=0x20000000,  length=0x00008000)                                           #Data region
+        )
+
+    def __init__(self, link):
+        super(TZ10xx, self).__init__(link, self.memoryMap)
+
+    def reset(self, software_reset=True):
+        if software_reset == None:
+            software_reset = True
+        super(TZ10xx, self).reset(software_reset)
+        if software_reset:
+            sleep(0.1)
+            self.__initializeDCDC()
+            self.__initializeVoltageMode()
+            self.__initializeClockSource()
+            self.__initializeAndPowerOnPLL()
+            self.__startUSB()
+            self.__initializePU()
+            self.__stopUSB()
+            self.__initializeClockSourceMain()
+            self.__initializePrescaler()
+            self.__powerOnDomain()
+            self.__initializeIOCell()
+            self.__resetOnDomain()
+            self.__powerOffDomain()
+            self.__stopPrescaler()
+            self.__initializeADPLL()
+            self.__initializeRTC()
+            self.__initializeMacro()
+            self.__initializeMiscRegisters()
```

Comment:
If the power button is held then this code is not needed. Maybe instead of customizing pyOCD so much here, you could have DAPLink keep power enabled when a debugger is connected. What do you think @f-okuhara? In future versions of pyOCD the goal is to pull target support from cmsis-packs. Because of this there will no longer be target customization like here since this detailed sequence might not be in cmsis-packs.